### ISSUE TITLE
verfication hangs and is loading forever at login

### ISSUE THREAD

**Issue by localguru** [vector-im/element-web#16420]
### Description

I upgraded synapse from 1.1.25 to 1.26.0 (Ubuntu packages from matrix.org, no workes) . Since then I can't verify the login for my user anymore. After entering the passphrase in element browser client (1.7.20 Firefox/Chrome) or element desktop client (1.7.20) "Verify this login" is loading for ever (tested on different plattforms). If I skip verification at login and do a "Connect this session to key backup" afterwards I can load all keys from backup with my passphrase and can decrypt encrypted messages. So the passphrase must be correct. The strange thing is, that this happens only with my userid, other userids on that homeserver seem not to be affected. I don't see any errors in synapse log, nor in nginx log.

See also https://github.com/matrix-org/synapse/issues/9352

### Steps to reproduce

- login via element browser or desktop client
- enter login password
- enter verification passphrase at "Verify this login"
- client is loading for ever

#### expected

Verification with valid passphrase should work.

Logs being sent: yes

**screenshot from browser console:**

![Bildschirmfoto_2021-02-09_14-22-33](https://user-images.githubusercontent.com/8544984/107393752-6d6b0d80-6afb-11eb-9a10-cf90a908dc80.png)

### Version information

- **Platform**: 
 - web (in broser) version 1.7.20
 - desktop version 1.7.20 on Ubuntu

For the web app:

- **Browser**:
- Firefox 85.0 (64-Bit)
- Chrome 88.0.4324.150 (Offizieller Build) (64-Bit)
- **OS**: Ubuntu focal
- **URL**: private web installation 1.7.20

For the desktop app:

- **OS**: Ubuntu focal
- **Version**:  Version Element: 1.7.20, Version olm: 3.2.1

Server version: synapse 1.26.0 on Ubuntu 18.04 LTS, packages from matrix.org, very vanilla installation, no workes, nginx as proxy.

**Comment by petrov-adg** [comment]
Same problem:
matrix-synapse-py3 1.26.0+focal1 ( Ubuntu 20.04 LTS )
NGINX as reverse proxy ( https://github.com/matrix-org/synapse/blob/develop/docs/reverse_proxy.md )

uncommenting public_baseurl in **homeserver.yaml** was solution (see https://github.com/matrix-org/synapse/issues/9264 )

**Comment by localguru** [comment]
@petrov-adg that is a different problem: if public_baseurl in homeserver.yaml is not set, you run in CORS errors and that's why login is not possible. My problem is different.

**Comment by localguru** [comment]
I did reset the "Cross-signing" and set the password to the same as the passphrase for key backup. If I enter this passphrase now after login at "Verify this login" everything is fine and works very well; I'm logged in and can read encrypted messages without "Connect this session to key backup" afterwards. Now for whatever reason I had to reset the password for cross-signing, the login should not hang in such a case. I will not close this ticket. Maybe someone else from element developer team will take a look at the problem. But feel free to close the ticket if you think there is no bug.
